router/router.go

```diff
@@ -197,7 +197,8 @@ func New(cfg *config.Configuration, rateConvertor *currency.RateConverter) (r *R
 	}
 
 	gvlVendorIDs := bidderInfos.ToGVLVendorIDMap()
-	gdprPerms := gdpr.NewPermissions(context.Background(), cfg.GDPR, gvlVendorIDs, generalHttpClient)
+	vendorListFetcher := gdpr.NewVendorListFetcher(context.Background(), cfg.GDPR, generalHttpClient, gdpr.VendorListURLMaker)
+	gdprPerms := gdpr.NewPermissions(cfg.GDPR, &cfg.GDPR.TCF2, gvlVendorIDs, vendorListFetcher)
```

Comment:
I started to make this change and then realized it is probably not worth it right now because I have another PR coming that will move the router GDPR object instantiation into the cookie sync and setuid endpoints, in which case the line `gvlVendorIDs := bidderInfos.ToGVLVendorIDMap()` in the router will no longer be executed twice (once in the router and once when the exchange is created) for the auction/amp/video endpoints.
As you mentioned, this shouldn't add any overhead so I'd rather not make that short-term change here if that's ok with you. This will be addressed though shortly :)